risc0/zkvm/sdk/rust/guest/src/alloc.rs

```diff
@@ -14,43 +14,259 @@
 
 use core::{
     alloc::{GlobalAlloc, Layout},
-    cell::UnsafeCell,
+    cmp::max,
+    mem::size_of,
+    ops::DerefMut,
 };
 
-use crate::{_fault, align_up, REGION_HEAP_END, REGION_HEAP_START, WORD_SIZE};
+use crate::mem_layout::{REGION_HEAP_END, REGION_HEAP_START};
+use crate::mutex::{Mutex, MutexGuard};
+use crate::sha::ShaBuf;
+use crate::{_fault, align_up, WORD_SIZE};
```

Comment:
For consistency, bundle uses:
```suggestion
use crate::{
  mem_layout::{REGION_HEAP_END, REGION_HEAP_START},
  mutex::{Mutex, MutexGuard},
  sha::ShaBuf,
  _fault, align_up, WORD_SIZE
};
```